artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java

```diff
@@ -2133,6 +2133,8 @@ public void destroyQueue(final SimpleString queueName,
          }
       }
 
+      getSecurityRepository().removeCacheMatch(queueName.toString());
```

Comment:
Please do not merge this yet.. I'm looking at some details.

Reply:
There is something wrong. We clear the entire cache every time a queue is removed.

This is actually being called twice.. on destroyQueue and on PostOffice.removeBinding.

Reply:
@jbertram ^^^